### PR TITLE
Add ASG cooldown and health_check_grace_period options

### PR DESCRIPTION
**Pull request by ArieLevs**
# PR o'clock

## Description

Add [default_cooldown](https://www.terraform.io/docs/providers/aws/r/autoscaling_group.html#default_cooldown)
and [health_check_grace_period](https://www.terraform.io/docs/providers/aws/r/autoscaling_group.html#health_check_grace_period)
options

### Checklist

- [ ] Change added to CHANGELOG.md. All changes must be added and breaking changes and highlighted
- [ ] CI tests are passing
- [ ] README.md has been updated after any changes to variables and outputs. See https://github.com/terraform-aws-modules/terraform-aws-eks/#doc-generation

### CODE REVIEW

**Comment by ArieLevs** [comment]
@barryib @max-rocket-internet 
Hi how can i get this PR get reviewed?

File: version [hunk 1]
```diff
@@ -1 +1 @@
-v8.2.0
+v8.2.1
```
Comment (max-rocket-internet) [inline version]: Please leave this file as it's only changed when a new release is created.

```suggestion
v8.2.0
```

File: workers_launch_template.tf [hunk 2]
```diff
@@ -72,6 +72,16 @@ resource "aws_autoscaling_group" "workers_launch_template" {
     "termination_policies",
     local.workers_group_defaults["termination_policies"]
   )
+  default_cooldown = lookup(
```
Comment (max-rocket-internet) [inline workers_launch_template.tf]: Can you please add these to `workers.tf` also?

**Comment by max-rocket-internet** [comment]
Screen Shot 2020-02-27 at 11 21 30

👈😃👈

**Comment by ArieLevs** [comment]
@max-rocket-internet 
Can you please look into this, 
I cannot find a way to resolve conflicts from my side without reverting the commit on changelog

**Comment by ArieLevs** [comment]
@max-rocket-internet 
Hi,
Anything i can do to promote this?

File: local.tf [hunk 3]
```diff
@@ -32,6 +32,8 @@ locals {
     asg_force_delete              = false                       # Enable forced deletion for the autoscaling group.
     asg_initial_lifecycle_hooks   = []                          # Initital lifecycle hook for the autoscaling group.
     asg_recreate_on_change        = false                       # Recreate the autoscaling group when the Launch Template or Launch Configuration change.
+    default_cooldown              = "300"                       # The amount of time, in seconds, after a scaling activity completes before another scaling activity can start.
```
Comment (barryib) [inline local.tf]: Where did you take the `300`  as default ? Maybe, the default should be `null` ?

Comment (ArieLevs) [reply local.tf]: As i checked the [health_check_grace_period](https://www.terraform.io/docs/providers/aws/r/autoscaling_group.html#health_check_grace_period) probably should be set to `null`,

But the [default_cooldown](https://www.terraform.io/docs/providers/aws/r/autoscaling_group.html#default_cooldown), is not being set by terraform (and should bee 300 as [stated here](https://docs.aws.amazon.com/autoscaling/ec2/userguide/Cooldown.html#cooldown-default)).
That means if set `default_cooldown` to lets say 90 seconds, and later remove this, the value will not get set back to `300` but stay on `90`.

i'm not sure who should enforce defaults (as for example [boto3](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/autoscaling.html#AutoScaling.Client.create_auto_scaling_group) has this default),

But let me know what should be the desired behaviour from this module? null?

Comment (max-rocket-internet) [reply local.tf]: I would go with null for both as this won't change any behaviour for existing users of the module. I think that is safest option?

Comment (ArieLevs) [reply local.tf]: updated

File: workers.tf [hunk 4]
```diff
@@ -73,6 +73,16 @@ resource "aws_autoscaling_group" "workers" {
     "termination_policies",
     local.workers_group_defaults["termination_policies"]
   )
+  default_cooldown = lookup(
+    var.worker_groups_launch_template[count.index],
```
Comment (max-rocket-internet) [inline workers.tf]: You have copy pasted something wrong here:

```suggestion
    var. worker_groups[count.index],
```

File: workers.tf [hunk 5]
```diff
@@ -73,6 +73,16 @@ resource "aws_autoscaling_group" "workers" {
...
+    "default_cooldown",
+    local.workers_group_defaults["default_cooldown"]
+  )
+  health_check_grace_period = lookup(
+    var.worker_groups_launch_template[count.index],
```
Comment (max-rocket-internet) [inline workers.tf]: You have copy pasted something wrong here:

```suggestion
    var.worker_groups[count.index],
```

**Comment by barryib** [comment]
@ArieLevs can you please update your branch and resolve conflits.

**Comment by ArieLevs** [comment]
this PR is a total mess containing other files, 
I'll just create a new one.

**Pull request closed without merging**